### PR TITLE
Feature/SOF-7254-2

### PR DESCRIPTION
**Pull request by VsevolodX**
For algorithmic generation of Standata entries in ESSE format from 1st source data

### CODE REVIEW

File: materials/sources/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    common_name: Graphene
+    lattice_type: HEX
+    space_group: P6/mmm
+    dimensionality: 2D
+    form_factor: Monolayer
+    source_id: 2dm-3993
+    source: 2dmatpedia
+    doi: 10.1038/s41597-019-0097-3
+    url: http://www.2dmatpedia.org/2dmaterials/doc/2dm-3993/
+  - filename: 2dm-4991.poscar
```
Comment (timurbazhirov) [inline materials/sources/manifest.yml:12]: Maybe add  a newline to make it easier to read?

File: pyproject.toml [hunk 2]
```diff
@@ -22,7 +22,7 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
-dev = ["pre-commit", "black", "ruff", "isort", "mypy", "pip-tools"]
+dev = ["pre-commit", "black", "ruff", "isort", "mypy", "pip-tools", "express-py"]
```
Comment (timurbazhirov) [inline pyproject.toml:25]: Readme should have explanation about generating JSON sources

File: create_materials.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            material_config = convert_to_esse(poscar)
+
+            name = f'{material_config["formula"]}, {source["common_name"]}, {source["lattice_type"]} ({source["space_group"]}) {source["dimensionality"]} ({source["form_factor"]}), {source["source_id"]}'
+
+            # slugify common name
+            common_name = source["common_name"].replace(' ', '_')
+            # create slugified filename
+            filename = f'{material_config["formula"]}-[{common_name}]-{source["lattice_type"]}_[{source["space_group"]}]_{source["dimensionality"]}_[{source["form_factor"]}]-[{source["source_id"]}]'
+
+            # replace special carachters with URL encoding
```
Comment (timurbazhirov) [inline create_materials.py:37]: characters

File: create_materials.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+def main():
+    materials = []
+    for source in read_manifest('materials/sources/manifest.yml'):
+        with open(f"materials/sources/{source['filename']}", 'r') as file:
+            poscar = file.read()
+            material_config = convert_to_esse(poscar)
+
+            name = f'{material_config["formula"]}, {source["common_name"]}, {source["lattice_type"]} ({source["space_group"]}) {source["dimensionality"]} ({source["form_factor"]}), {source["source_id"]}'
+
+            # slugify common name
```
Comment (timurbazhirov) [inline create_materials.py:32]: the common name

File: create_materials.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            common_name = source["common_name"].replace(' ', '_')
+            # create slugified filename
+            filename = f'{material_config["formula"]}-[{common_name}]-{source["lattice_type"]}_[{source["space_group"]}]_{source["dimensionality"]}_[{source["form_factor"]}]-[{source["source_id"]}]'
+
+            # replace special carachters with URL encoding
+            filename = filename.replace('/', '%2F')
+
+            material_config["name"] = name
+
+            # remove keys apart from name, lattice, basis.
```
Comment (timurbazhirov) [inline create_materials.py:42]: remove all keys except for ...

File: materials/sources/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    lattice_type: HEX
+    space_group: P6/mmm
+    dimensionality: 2D
+    form_factor: Monolayer
+    source_id: 2dm-4991
+    source: 2dmatpedia
+    doi: 10.1038/s41597-019-0097-3
+    url: http://www.2dmatpedia.org/2dmaterials/doc/2dm-4991/
+
+
```
Comment (timurbazhirov) [inline materials/sources/manifest.yml:24]: Fix the linter, pls

File: pyproject.toml [hunk 7]
```diff
@@ -22,7 +22,7 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
-dev = ["pre-commit", "black", "ruff", "isort", "mypy", "pip-tools"]
+dev = ["pre-commit", "black", "ruff", "isort", "mypy", "pip-tools", "express-py", "pyyaml"]
```
Comment (timurbazhirov) [inline pyproject.toml:25]: Let's add some explanation in README about how to create the files

File: README.md [hunk 8]
```diff
@@ -94,11 +95,59 @@ ts-node src/js/cli.ts -d tmp materials/categories.yml
...
 
-
 ## 4. Development
 
 See [ESSE](https://github.com/Exabyte-io/esse) for the notes about development and testing.
 
+### 4.1. Materials Source
+
+The materials data is sourced from the [Materials Project](https://next-gen.materialsproject.org) for 3D materials and [2dmatpedia](http://www.2dmatpedia.org) for 2D materials. The structural data in POSCAR format is stored in the `materials/sources` directory alongside the `manifest.yml` file that contains the additional description and metadata for each material.
+Python script `create_materials.py` is used to create the materials data files from the source data.
```
Comment (timurbazhirov) [inline README.md:105]: How exactly is it supposed to be used?? Put commands below

File: create_materials.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
+import yaml
+import json
+from express import ExPrESS
+
+def read_manifest(manifest_path):
```
Comment (timurbazhirov) [inline create_materials.py:5]: Use types declarations for the variables in py

File: pyproject.toml (hunk 7 above)
Comment (timurbazhirov) [reply pyproject.toml:25]: Pyyaml is already above in `dependencies`, it should not be needed here. If you do `pip install ".[dev]"` it should pick it up

**Review by timurbazhirov — APPROVED** [review]

File: README.md [hunk 10]
```diff
@@ -94,11 +95,64 @@ ts-node src/js/cli.ts -d tmp materials/categories.yml
...
 See [ESSE](https://github.com/Exabyte-io/esse) for the notes about development and testing.
 
+### 4.1. Materials Source
+
+The materials data is sourced from the [Materials Project](https://next-gen.materialsproject.org) for 3D materials and [2dmatpedia](http://www.2dmatpedia.org) for 2D materials. The structural data in POSCAR format is stored in the `materials/sources` directory alongside the `manifest.yml` file that contains the additional description and metadata for each material.
+
+To add new materials to Standata, place the POSCAR file in the `materials/sources` directory and update the `manifest.yml` file with the new material's metadata. Then run to create the materials data:
+
+ ```shell
+ python create_materials.py
```
Comment (timurbazhirov) [inline README.md:109]: How about installing dependencies? We prob need `pip install ".[dev]"` and a virtualenv.

File: create_materials.py [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+    filename = filename.replace('/', '%2F')
+
+    return filename
+
+def main():
+    """
+    Main function to create materials listed in the sources manifest.
+    """
+    materials = []
+    for source in read_manifest('materials/sources/manifest.yml'):
```
Comment (timurbazhirov) [inline create_materials.py:94]: Move the path to MANIFEST_PATH var at the top

File: create_materials.py [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+        with open(f"materials/sources/{source['filename']}", 'r') as file:
```
Comment (timurbazhirov) [inline create_materials.py:95]: SOURCES_PATH

File: create_materials.py [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+                "source": source["source"],
+                "doi": source["doi"],
+                "url": source["url"],
+                "origin": True
+            }
+
+            # add "isNonPeriodic" property
+            material_config['isNonPeriodic'] = False
+
+            with open(f'materials/{filename}.json', 'w') as file:
```
Comment (timurbazhirov) [inline create_materials.py:118]: TARGETS_PATH or DESTINATION_PATH

**Review by timurbazhirov — APPROVED** [review]

File: pyproject.toml (hunk 7 above)
Comment (VsevolodX) [reply pyproject.toml:25]: Yes, it does.

File: README.md (hunk 10 above)
Comment (VsevolodX) [reply README.md:109]: I'll place that above as it's the general dev guidance

**Review by timurbazhirov — APPROVED** [review]

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**